### PR TITLE
fix points api signature

### PR DESCRIPTION
**Pull request by transphorm**
## Release Notes

* **New Features**
  * Enhanced Points API with intelligent key selection strategy for improved signing reliability.

* **Bug Fixes**
  * Fixed referral flow to execute once per referral, preventing duplicate executions.
  * Referrer now clears when error modal is dismissed, preventing potential retry loops.

* **Tests**
  * Expanded Points API test coverage for signing scenarios and edge cases.
  * Updated referral confirmation tests to verify one-time execution guard.

### CODE REVIEW

File: app/src/utils/points/api.ts [hunk 1]
```diff
@@ -49,8 +53,23 @@ export const isSuccessfulStatus = (status: number): boolean =>
...
+    let privateKey: string | null = null;
+    try {
+      const pointsAddr = (await getPointsAddress()).toLowerCase();
+      if (signingAddress === pointsAddr) {
+        privateKey = await unsafe_getPointsPrivateKey();
+      }
+    } catch {
+      // If fetching the points address fails for any reason, fall back to primary key
+    }
+    if (!privateKey) {
```
Comment (remicolin) [inline app/src/utils/points/api.ts]: are we sure we want to use the privateKey here
